pkg/deploy/deploy.go

```diff
@@ -210,21 +212,22 @@ func manageResource(ctx context.Context, cli client.Client, obj *unstructured.Un
 
 	// Return if error getting resource in cluster
 	found, err := getResource(ctx, cli, obj)
-	if err != nil {
+	if client.IgnoreNotFound(err) != nil {
 		return err
 	}
 
-	if !enabled {
+	// err == nil means found
+	if !enabled && err == nil {
 		return handleDisabledComponent(ctx, cli, found, componentName)
 	}
 
-	// Create resource if it doesn't exist
-	if found == nil {
-		return createResource(ctx, cli, obj, owner)
+	if err == nil {
```

Comment:
should this be `if found == nil` ?

Reply:
I've exchanged createResource/updateResource to avoid found == nil check

Reply:
sorry, meant to say 
if found != nil { // do update }

Reply:
This will work as well, but if there was no error (err == nil), found is not nil in getResource() so we can stay on checking err only

Reply:
This could be more clear actually:
```
func manageResource(ctx context.Context, cli client.Client, obj *unstructured.Unstructured, owner metav1.Object, applicationNamespace, componentName string, enabled bool) error {
	// Return if resource is of Kind: Namespace and Name: odhApplicationsNamespace
	if obj.GetKind() == "Namespace" && obj.GetName() == applicationNamespace {
		return nil
	}

	found, err := getResource(ctx, cli, obj)

	// err == nil means found
	if err == nil {
		if enabled {
			return updateResource(ctx, cli, obj, found, owner, componentName)
		}
		return handleDisabledComponent(ctx, cli, found, componentName)
	}

	if apierrs.IsNotFound(err) {
		// Create resource if it doesn't exist
		return createResource(ctx, cli, obj, owner)
	}

	return err
}
```